src/youtube-tracker.coffee

```diff
@@ -14,9 +13,43 @@
 # Author:
 #   Karl Birch <mbwkarl@gmail.com>
 
+GOOGLE_API_KEY = process.env.GOOGLE_API_KEY
+
+querystring = require 'querystring'
+url = require 'url'
+
 module.exports = (robot) ->
-  robot.respond /hello/, (res) ->
-    res.reply "hello!"
+  if not GOOGLE_API_KEY?
+    return robot.logger.error "hubot-youtube-tracker: Missing GOOGLE_API_KEY in environment.  Please set and try again."
+
+  robot.respond /(?:tracker|ytt) recent (.*)/i, id: "tracker.recent", (res) ->
+    query = res.match[1]
+    # get channel
+    QueryYoutube res, '/channels', { part: 'contentDetails', forUsername: query }, (channelObject) ->
+      if not channelObject.items[0]
+        res.send "No channel found"
+        return
+      uploadsId = channelObject.items[0].contentDetails.relatedPlaylists.uploads
+      # get playlist
+      QueryYoutube res, '/playlistItems', { part: 'snippet', playlistId: uploadsId, maxResults: 3 }, (playlistObject) ->
+        videos = []
+        for item in playlistObject.items
+          videos.push( { title: item.snippet.title, id: item.snippet.resourceId.videoId } )
+        resstr = ""
+        for v in videos
+          resstr += v.title + " | https://youtu.be/" + v.id + '\n'
```

Comment:
Could we not combine this loop with the one before? It's essentially:

```coffee
resstr += item.snippet.title + " | https://youtu.be/" + item.snippet.resourceId.videoId + '\n'
```

On that note, we can use CoffeeScript's Ruby-style string interpolation, e.g.

```coffee
resstr += "#{item.snippet.title} | https://youtu.be/#{item.snippet.resourceId.videoId}\n"
```